### ISSUE TITLE
"Perform Action" action in Shortcuts is missing in 2024.5 Update

### ISSUE THREAD

**Issue by yur1xpp** [home-assistant/iOS#2791]
**iOS device model, version and app version**

Model Name: iPhone 14 Pro
Software Version: iOS 17.0
App version: 2024.5

**Home Assistant Core Version**

2024.5.5

**Describe the bug**
The "Perform Action" action is missing in 2024.5 update. Reverting to 2024.4 fixed this problem temporally. I'm actually quite surprise that no one mention this yet, since I used this action in almost all of my HA shortcuts.

**To Reproduce**
Open Shortcuts app, search Home Assistant, notice "Perform Action" is missing.

**Expected behavior**
"Perform Action" action exists, like in 2024.4.

**Screenshots**

**Additional context**

**Comment by bgoncal** [comment]
That's weird, I do see it in shortcuts, did you update your iOS recently? Try to reboot your iPhone to see if it comes up, otherwise let me know

**Comment by yur1xpp** [comment]
Hey @bgoncal thanks for the reply. Indeed it's very weird. I actually did reboot the device. Here are the two screenshots, first one is 2024.4, and second is 2024.5.

![2024.4](https://github.com/home-assistant/iOS/assets/34767893/e0f12be1-9a6b-4ca2-8675-dd1bbf5947d5)


![2024.5](https://github.com/home-assistant/iOS/assets/34767893/e296f923-0145-4e9e-b911-0f93eecdd131)

Does this have anything to do with me being on iOS 17.0? Hopefully not, since I would like to stay on this iOS version for OS unlocking reasons.

**Comment by bgoncal** [comment]
> Hey @bgoncal thanks for the reply. Indeed it's very weird. I actually did reboot the device. Here are the two screenshots, first one is 2024.4, and second is 2024.5.
> 
> ![2024.4](https://github.com/home-assistant/iOS/assets/34767893/e0f12be1-9a6b-4ca2-8675-dd1bbf5947d5)
> 
> 
> ![2024.5](https://github.com/home-assistant/iOS/assets/34767893/e296f923-0145-4e9e-b911-0f93eecdd131)
> 
> Does this have anything to do with me being on iOS 17.0? Hopefully not, since I would like to stay on this iOS version for OS unlocking reasons.

It shouldn't matter, we have a change between iOS 17 and before it but since 2024.4 works on iOS 17 for you that's shouldn't be related.

I'll check on Monday if I can find anything that could be related to that, when you say "OS unlocking" you mean jailbreak?

**Comment by yur1xpp** [comment]
> It shouldn't matter, we have a change between iOS 17 and before it but since 2024.4 works on iOS 17 for you that's shouldn't be related.

I'm glads that's the case.

> I'll check on Monday if I can find anything that could be related to that, when you say "OS unlocking" you mean jailbreak?

Thanks! You're perfectly right, though right now the device is not yet jailbroken (hopefully soon), so it shouldn't cause any conflicts in this ticket.

**Comment by yur1xpp** [comment]
Just got the TestFlight version installed. Seems like the action does appear, but it throws an error whenever trying to run or select actions.

![image](https://github.com/home-assistant/iOS/assets/34767893/0e7e36b9-f453-40b6-9ff9-0f04add44352)

**Comment by bgoncal** [comment]
I see that there is no action selected, could you choose one and try again?

**Comment by yur1xpp** [comment]
Unfortunately that error appears during the selection as well, as you tap it the error appears.

**Comment by bgoncal** [comment]
Did you configure your actions in YAML or in the App? Can you share with me the configuration? (It if is in the app a screenshot of each one is already enough)

**Comment by yur1xpp** [comment]
I configured them in the app itself. Here are some of the screenshots.

![image](https://github.com/home-assistant/iOS/assets/34767893/dff0a4ff-90da-488b-83d8-667720197c3e) 

![image](https://github.com/home-assistant/iOS/assets/34767893/54deacf4-d3ab-40a2-8e04-225d97564ac5)

![image](https://github.com/home-assistant/iOS/assets/34767893/e6163b54-8325-4da3-884e-b5016a765ff1)

**Comment by yur1xpp** [comment]
> I still couldn't find a way to reproduce the issue, do you keep experiencing this?

Unfortunately I'm still experiencing this even on 2024.6 (2024.693). Downgrading to 2024.4 fixes this every single time.

I'm hesitant to attach the log here as it seems it contains so much private data including precise location and identifiable information, but I would love to help you find the relevant info that I can attach here. Looking at the intent logs, it seems quite normal to me. Please let me know if you need me to search for specific keywords.

**Comment by bgoncal** [comment]
You can send me in discord if you prefer (bgoncal2), but I've been googling a bit and it looks like this error happens before even reaching the App, which is a big problem because if thats true thats not something we could fix and since you want to stay on iOS 17 I can't suggest you to upgrade to have it fixed.

The only change related to "Perform Action" happened in 2024.3, where we have an implementation for newer OS's (iOS17+) and older (Siri Intents/App Intents), but since 2024.4 works for you Im not really sure what may be causing it, no other version after 2024.4 works right?

**Comment by yur1xpp** [comment]
> looks like this error happens before even reaching the App

This is so weird since it works great in previous versions. Will try to take my time to redact any sensitive information prior sending. Gotta find some time to do this though, so might not be so soon.

> The only change related to "Perform Action" happened in 2024.3, where we have an implementation for newer OS's (iOS17+) and older (Siri Intents/App Intents), but since 2024.4 works for you Im not really sure what may be causing it, no other version after 2024.4 works right?

That's correct that no any other version after 2024.4 works for me. One thing that I noticed is that in 2024.2, the icon of the actions showed in the shortcuts. And in 2024.3 and after, it completely disappeared. Is this by design or a regression of the update? Will this be the cause of the issue that I'm experiencing in >2024.4?

![2024.2](https://github.com/home-assistant/iOS/assets/34767893/6c82bb8e-337e-4954-871d-ded2784e364c)

![2024.3](https://github.com/home-assistant/iOS/assets/34767893/fe1fd8dd-036c-4c42-abc8-2cb4a5fb9f08)

**Comment by bgoncal** [comment]
Good catch, the icons should be re-added then, but I don't see a relation to why it fails to perform action

**Comment by yur1xpp** [comment]
I'm curious, do the icons appears on your devices? Perhaps the one possible reason is that it failed to "populate" or "parse" the icons and thus making it error out? I do want to clarity that, the "Perform Action" failed at even displaying the list of actions in >2024.4, i.e tapping the "Action" will throw out the error we discussed.
![image](https://github.com/home-assistant/iOS/assets/34767893/c2f4bb80-3f74-4628-a8b6-8a7b56914e79)

**Comment by bgoncal** [comment]
I dont see the icons either, but thats not related, it was just something left out a few releases ago.

**Comment by bgoncal** [comment]
Are you able to add a widget that runs an action to your Home Screen?

**Comment by bgoncal** [comment]
Also, let's try another approach. Can you create those actions via configuration.yaml ? Then I can copy and paste and try on my side too on an iOS 17.0 simulator

**Comment by yur1xpp** [comment]
@bgoncal I first must thank you for your patience with me in this long issue.

> Are you able to add a widget that runs an action to your Home Screen?

Unfortunately I cannot, the same _The action "Actions" could not run because an internal error occurred_ got thrown during the widget editing.  I did this in v2024.6.

> Also, let's try another approach. Can you create those actions via configuration.yaml ? Then I can copy and paste and try on my side too on an iOS 17.0 simulator

I just recreate all my actions in the yaml file. The error still happens unfortunately. I deleted all my actions created in the iOS app, and the only actions exist are the newly created in yaml file. Also tested in v2024.6. These are my list of actions:

```
ios:
  actions:
    - name: Apple Watch Charger
      background_color: "#000000"
      label:
        text: "Apple Watch Charger"
        color: "#ff0000"
      icon:
        icon: watch-variant
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: On Apple Watch Charger
      background_color: "#000000"
      label:
        text: "On Apple Watch Charger"
        color: "#ff0000"
      icon:
        icon: watch-variant
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: Off Apple Watch Charger
      background_color: "#000000"
      label:
        text: "Off Apple Watch Charger"
        color: "#ff0000"
      icon:
        icon: watch-variant
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: Lock Predator
      background_color: "#000000"
      label:
        text: "Lock Predator"
        color: "#ff0000"
      icon:
        icon: account-lock
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: iPad Pro Charger
      background_color: "#000000"
      label:
        text: "iPad Pro Charger"
        color: "#ff0000"
      icon:
        icon: tablet
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: On iPad Pro Charger
      background_color: "#000000"
      label:
        text: "On iPad Pro Charger"
        color: "#ff0000"
      icon:
        icon: tablet
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: Off iPad Pro Charger
      background_color: "#000000"
      label:
        text: "Off iPad Pro Charger"
        color: "#ff0000"
      icon:
        icon: tablet
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: MagSafe Charger
      background_color: "#000000"
      label:
        text: "MagSafe Charger"
        color: "#ff0000"
      icon:
        icon: cellphone
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: On MagSafe Charger
      background_color: "#000000"
      label:
        text: "On MagSafe Charger"
        color: "#ff0000"
      icon:
        icon: cellphone
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: Off MagSafe Charger
      background_color: "#000000"
      label:
        text: "Off MagSafe Charger"
        color: "#ff0000"
      icon:
        icon: cellphone
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: Dell Monitor
      background_color: "#000000"
      label:
        text: "Dell Monitor"
        color: "#ff0000"
      icon:
        icon: monitor
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: On Dell Monitor
      background_color: "#000000"
      label:
        text: "On Dell Monitor"
        color: "#ff0000"
      icon:
        icon: monitor
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: Off Dell Monitor
      background_color: "#000000"
      label:
        text: "Off Dell Monitor"
        color: "#ff0000"
      icon:
        icon: monitor
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: Run Chargers Automation
      background_color: "#000000"
      label:
        text: "Run Chargers Automation"
        color: "#ff0000"
      icon:
        icon: battery-charging-medium
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: Kitchen Fan
      background_color: "#000000"
      label:
        text: "Kitchen Fan"
        color: "#ff0000"
      icon:
        icon: fan
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: On Kitchen Fan
      background_color: "#000000"
      label:
        text: "On Kitchen Fan"
        color: "#ff0000"
      icon:
        icon: fan
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: Off Kitchen Fan
      background_color: "#000000"
      label:
        text: "Off Kitchen Fan"
        color: "#ff0000"
      icon:
        icon: fan
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: amazingTHING
      background_color: "#000000"
      label:
        text: "amazingTHING"
        color: "#ff0000"
      icon:
        icon: cellphone
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: On amazingTHING
      background_color: "#000000"
      label:
        text: "On amazingTHING"
        color: "#ff0000"
      icon:
        icon: cellphone
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: Off amazingTHING
      background_color: "#000000"
      label:
        text: "Off amazingTHING"
        color: "#ff0000"
      icon:
        icon: cellphone
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: Master Bedroom Aircond
      background_color: "#000000"
      label:
        text: "Master Bedroom Aircond"
        color: "#ff0000"
      icon:
        icon: air-conditioner
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: On Master Bedroom Aircond
      background_color: "#000000"
      label:
        text: "On Master Bedroom Aircond"
        color: "#ff0000"
      icon:
        icon: air-conditioner
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: Off Master Bedroom Aircond
      background_color: "#000000"
      label:
        text: "Off Master Bedroom Aircond"
        color: "#ff0000"
      icon:
        icon: air-conditioner
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: Workstation Aircond
      background_color: "#000000"
      label:
        text: "Workstation Aircond"
        color: "#ff0000"
      icon:
        icon: air-conditioner
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: On Workstation Aircond
      background_color: "#000000"
      label:
        text: "On Workstation Aircond"
        color: "#ff0000"
      icon:
        icon: air-conditioner
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: Off Workstation Aircond
      background_color: "#000000"
      label:
        text: "Off Workstation Aircond"
        color: "#ff0000"
      icon:
        icon: air-conditioner
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: Porch Light
      background_color: "#000000"
      label:
        text: "Porch Light"
        color: "#ff0000"
      icon:
        icon: lightbulb-fluorescent-tube
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: On Porch Light
      background_color: "#000000"
      label:
        text: "On Porch Light"
        color: "#ff0000"
      icon:
        icon: lightbulb-fluorescent-tube
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: Off Porch Light
      background_color: "#000000"
      label:
        text: "Off Porch Light"
        color: "#ff0000"
      icon:
        icon: lightbulb-fluorescent-tube
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: Master Bedroom Fan
      background_color: "#000000"
      label:
        text: "Master Bedroom Fan"
        color: "#ff0000"
      icon:
        icon: fan
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: On Master Bedroom Fan
      background_color: "#000000"
      label:
        text: "On Master Bedroom Fan"
        color: "#ff0000"
      icon:
        icon: fan
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: Off Master Bedroom Fan
      background_color: "#000000"
      label:
        text: "Off Master Bedroom Fan"
        color: "#ff0000"
      icon:
        icon: fan
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: Master Bedroom Humidifier
      background_color: "#000000"
      label:
        text: "Master Bedroom Humidifier"
        color: "#ff0000"
      icon:
        icon: air-humidifier
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: On Master Bedroom Humidifier
      background_color: "#000000"
      label:
        text: "On Master Bedroom Humidifier"
        color: "#ff0000"
      icon:
        icon: air-humidifier
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: Off Master Bedroom Humidifier
      background_color: "#000000"
      label:
        text: "Off Master Bedroom Humidifier"
        color: "#ff0000"
      icon:
        icon: air-humidifier
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: Workstation Humidifier
      background_color: "#000000"
      label:
        text: "Workstation Humidifier"
        color: "#ff0000"
      icon:
        icon: air-humidifier
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: On Workstation Humidifier
      background_color: "#000000"
      label:
        text: "On Workstation Humidifier"
        color: "#ff0000"
      icon:
        icon: air-humidifier
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: Off Workstation Humidifier
      background_color: "#000000"
      label:
        text: "Off Workstation Humidifier"
        color: "#ff0000"
      icon:
        icon: air-humidifier
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: Bedroom Comfy
      background_color: "#000000"
      label:
        text: "Bedroom Comfy"
        color: "#ff0000"
      icon:
        icon: bed-queen
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: On Bedroom Comfy
      background_color: "#000000"
      label:
        text: "On Bedroom Comfy"
        color: "#ff0000"
      icon:
        icon: bed-queen
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: Off Bedroom Comfy
      background_color: "#000000"
      label:
        text: "Off Bedroom Comfy"
        color: "#ff0000"
      icon:
        icon: bed-queen
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: Workstation Comfy
      background_color: "#000000"
      label:
        text: "Workstation Comfy"
        color: "#ff0000"
      icon:
        icon: desktop-tower-monitor
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: On Workstation Comfy
      background_color: "#000000"
      label:
        text: "On Workstation Comfy"
        color: "#ff0000"
      icon:
        icon: desktop-tower-monitor
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: Off Workstation Comfy
      background_color: "#000000"
      label:
        text: "Off Workstation Comfy"
        color: "#ff0000"
      icon:
        icon: desktop-tower-monitor
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
    - name: Kill Spotify
      background_color: "#000000"
      label:
        text: "Kill Spotify"
        color: "#ff0000"
      icon:
        icon: spotify
        color: "#ffffff"
      show_in_carplay: false
      show_in_watch: false
```

**Comment by yur1xpp** [comment]
And for another test result (now that actions are available across all devices), on the same v2024.6, but this time in iOS 16.3.1. I got error as well, but it's _An unknown error occurred_. I suspect this might not be device/iOS version specific issue anymore, but rather maybe the actions list too long and it timed out?

![New Shortcut](https://github.com/home-assistant/iOS/assets/34767893/600502d3-1af5-463b-bdc1-704929f12429)

**Comment by bgoncal** [comment]
Good news is, I was able to reproduce the issue specifically in iOS 17 (using simulator), also tested on latest iOS (17.4.1) and on it it works fine.

I'll debug now to see if I can make it work also on iOS 17

**Comment by bgoncal** [comment]
Unfortunately I don't think a fix will be possible, it looks like 17.0 is the problem here and you are in a not so nice position because Siri Intents (previously intents method used) only works automatically for iOS versions before iOS 16 and App Intents (which has a problem in iOS 17.0) automatically is preferred after iOS 16.

Previous versions of the app may not display the issue because they were only using Siri Intents, which we can't move forward with them since they don't allow background execution in widgets.

I know it's not the answer you want but I would recommend updating your iOS

**Comment by yur1xpp** [comment]
That's what I'm afraid to hear since 17.0 getting some form of jailbreak is very close 😢 But I do understands that. 

I might have to ditch the "Perform Action" shortcuts action and migrate to "Call Service" manually (hopefully same issue won't happen for call service). iOS 17.0 is just too precious for my personal usage.

Thanks so much for helping me debugging this though!!
